template/src/core/network/async-process/asyncProcess.d.ts

```diff
@@ -0,0 +1,26 @@
+interface AsyncProcessState<Data = any, Payload = any> {
+  isRequestPending: boolean;
+  isRequestFetched: boolean;
+  data: Data | null;
+  error:
+    | (Error & {
+        data: any;
+        status: number;
+      })
+    | null;
+  requestPayload?: Payload;
+}
+
+interface UseAsyncProcessOptions<Data = any> {
+  initialState?: AsyncProcessState<Data>;
+  shouldResetDataWhenPending?: boolean;
+}
+
+type AsyncProcessCallBack<Data extends any> = <Response extends Data>(
+  promise: Promise<Response>,
+  responseSerializer?: (response: Response) => Data
+) => Promise<Response>;

```

Comment:
```suggestion
type AsyncProcessCallBack<Data extends any> = <Response extends Data>(
  promise: Promise<Response>,
  options?: {
    forceResetPreviousAsyncState?: boolean;
    responseSerializer?: (response: Response) => Data;
  }
) => Promise<Response>;
```

You can check Tinyman for reference. I think we should be able to control if a specific call to `run` function should reset the previous state or not, regardless of what was provided when calling `useAsyncProcess`. We can discuss about the `responseSerializer` option, I never used it yet and its type is probably wrong.